funcx_endpoint/funcx_endpoint/endpoint/rabbit_mq/result_queue_publisher.py

```diff
@@ -0,0 +1,108 @@
+import logging
+import time
+
+import pika
+
+logger = logging.getLogger(__name__)
+
+
+class ResultQueuePublisher:
+    """ResultPublisher publishes results to a topic exchange, with the {endpoint_id}.results
+    as a routing key.
+    """
+
+    def __init__(
+        self,
+        endpoint_id: str,
+        pika_conn_params: pika.ConnectionParameters,
+        exchange="results",
+    ):
+        self.endpoint_id = endpoint_id
+        self.routing_key = f"{self.endpoint_id}.results"
+        self.params = pika_conn_params
+        self.params.heartbeat = 30
+        self.params.blocked_connection_timeout = 60
+        self.exchange = exchange
+        self.exchange_type = "topic"
+        self._is_connected = False
+        self._deliveries = None
+        self._channel = None
+        self._stopping = False
+
+    def connect(self) -> pika.BlockingConnection:
```

Comment:
Rather than having `ResultQueuePublisher` responsible for constructing its own connections, I think it would be better to make `pika.BlockingConnection`. That would eliminate any need for `pika_conn_params`, so it simplifies. Plus it's more flexible: e.g. a mock connection could be passed for testing.

The same holds for the other classes which are wrapping connections. e.g. `ResultQueueSubscriber` could take a `pika.SelectConnection`.